### ISSUE TITLE
Error: exec /usr/local/bin/docker-entrypoint.sh: no such file or directory

### ISSUE THREAD

**Issue by paulodiovani** [docker-library/redis#327]
## Description

The docker image `redis:7.0.4-alpine` (current latest) published to docker registry is missing the entrypoint file.

```bash
$ docker run --rm -p 6379:6379 redis:7.0.4-alpine
exec /usr/local/bin/docker-entrypoint.sh: no such file or directory
```

The source in this repository includes it (https://github.com/docker-library/redis/tree/5c0bbb5/7.0/alpine) and I can see the copy command in the image history. And the file exists in the correct place.

```bash
$ docker history redis:7.0.4-alpine
IMAGE          CREATED       CREATED BY                                      SIZE      COMMENT
9192ed4e4955   2 weeks ago   /bin/sh -c #(nop)  CMD ["redis-server"]         0B
<missing>      2 weeks ago   /bin/sh -c #(nop)  EXPOSE 6379                  0B
<missing>      2 weeks ago   /bin/sh -c #(nop)  ENTRYPOINT ["docker-entry…   0B
<missing>      2 weeks ago   /bin/sh -c #(nop) COPY file:a9e7249f657e2eec…   664B
<missing>      2 weeks ago   /bin/sh -c #(nop) WORKDIR /data                 0B
<missing>      2 weeks ago   /bin/sh -c #(nop)  VOLUME [/data]               0B
<missing>      2 weeks ago   /bin/sh -c mkdir /data && chown redis:redis …   0B
<missing>      2 weeks ago   /bin/sh -c set -eux;   apk add --no-cache --…   21.6MB
<missing>      2 weeks ago   /bin/sh -c #(nop)  ENV REDIS_DOWNLOAD_SHA=f0…   0B
<missing>      2 weeks ago   /bin/sh -c #(nop)  ENV REDIS_DOWNLOAD_URL=ht…   0B
<missing>      2 weeks ago   /bin/sh -c #(nop)  ENV REDIS_VERSION=7.0.4      0B
<missing>      2 weeks ago   /bin/sh -c apk add --no-cache   'su-exec>=0.…   1.34MB
<missing>      2 weeks ago   /bin/sh -c addgroup -S -g 1000 redis && addu…   4.7kB
<missing>      2 weeks ago   /bin/sh -c #(nop)  CMD ["/bin/sh"]              0B
<missing>      2 weeks ago   /bin/sh -c #(nop) ADD file:2a949686d9886ac7c…   5.54MB
```

```bash
$ docker run --rm -it --entrypoint='' redis:7.0.4-alpine sh
sh-5.1# ls -l /usr/local/bin/docker-entrypoint.sh
-rwxrwxr-x 1 root root 664 Aug  9 22:51 /usr/local/bin/docker-entrypoint.sh
```

Regardless, the error still happens.

## Possible cause

Trying to run the entrypoint manually, it gives a more detailed error, that the interpreter `/bin/sh` is missing, which we can confirm.

```bash
sh-5.1# docker-entrypoint.sh
sh: /usr/local/bin/docker-entrypoint.sh: /bin/sh: bad interpreter: No such file or directory
sh-5.1# ls -l /bin/sh
ls: cannot access '/bin/sh': No such file or directory
```

Note: `/bin/sh` was present in image `redis:7.0.3-alpine`

## Suggested solution

Replace the shebang with `#!/usr/bin/sh` in `docker-entrypoint.sh`

**Comment by paulodiovani** [comment]
I thought the issue could have been caused by the `alpine` version, bit `alpine:3.16` have `/bin/sh` in place, so I don't have a strong clue on what could have caused the issue.

Maybe just rebuilding and republish will fix it.

**Comment by paulodiovani** [comment]
Quick way to test the issue:

```
$ docker run --rm --entrypoint='' redis:7.0.4-alpine ls -l /bin/sh
ls: cannot access '/bin/sh': No such file or directory

```

**Comment by wglambert** [comment]
I'm not able to reproduce
```console
$ docker run -d --name redis redis:7.0.4-alpine
66558e4bc8f03cef7a347a27264189229f6bb318b905b9ee5750870d82d4e698

$ docker exec -it redis ash

/data # ls /usr/local/bin
docker-entrypoint.sh  redis-check-aof       redis-cli             redis-server
redis-benchmark       redis-check-rdb       redis-sentinel

/data # redis-cli
127.0.0.1:6379> acl whoami
"default"
```

Using your first command:
```console
$ docker run --rm -p 6379:6379 redis:7.0.4-alpine
1:C 24 Aug 2022 17:27:43.550 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
1:C 24 Aug 2022 17:27:43.550 # Redis version=7.0.4, bits=64, commit=00000000, modified=0, pid=1, just started
1:C 24 Aug 2022 17:27:43.550 # Warning: no config file specified, using the default config. In order to specify a config file use redis-server /path/to/redis.conf
1:M 24 Aug 2022 17:27:43.551 * monotonic clock: POSIX clock_gettime
1:M 24 Aug 2022 17:27:43.552 * Running mode=standalone, port=6379.
1:M 24 Aug 2022 17:27:43.553 # WARNING: The TCP backlog setting of 511 cannot be enforced because /proc/sys/net/core/somaxconn is set to the lower value of 128.
1:M 24 Aug 2022 17:27:43.553 # Server initialized
1:M 24 Aug 2022 17:27:43.553 * Ready to accept connections
```

Your other test
```console
$ docker run --rm --entrypoint='' redis:7.0.4-alpine ls -l /bin/sh
lrwxrwxrwx    1 root     root            12 Jul 18 15:34 /bin/sh -> /bin/busybox
```

**Comment by paulodiovani** [comment]
A colleague from my company can't either. :confused: 
Looks like something in my machine, but I can't tell what could affect it.

**Comment by wglambert** [comment]
Was the image built locally but without the entrypoint? You could try doing a `docker image rm -f redis:7.0.4-alpine`

**Comment by paulodiovani** [comment]
> Was the image built locally but without the entrypoint? You could try doing a `docker image rm -f redis:7.0.4-alpine`

No, it was pulled, from the registry.
But deleting all images and pulling again fixed it.

**Comment by paulodiovani** [comment]
Ok, problem solved. Thanks for all the help.

```
$ docker run --rm --entrypoint='' redis:alpine ls -l /bin/sh
lrwxrwxrwx    1 root     root            12 Aug  9 08:47 /bin/sh -> /bin/busybox
```

I still don't know what caused the issue,
I suspect it could have been due to brtfs + timeshift (I restored a snapshot recently). Somehow I ended up with an incomplete or wrong (missing a symlink, to be precise) image identified by `9192ed4e4955`.

Note: the image wasn't pulled today, I already had it and I suspect it was working before.

Deleting all the layers (as @wglambert suggested) and pulling again fixed the issue.
